cluster-autoscaler/cloudprovider/mcm/mcm_manager.go

```diff
@@ -81,10 +83,20 @@ type McmManager struct {
 	namespace               string
 	interrupt               chan struct{}
 	discoveryOpts           cloudprovider.NodeGroupDiscoveryOptions
-	machineclient           machineapi.MachineV1alpha1Interface
-	coreclient              kubernetes.Interface
+	machineClient           machineapi.MachineV1alpha1Interface
+	coreClient              kubernetes.Interface
 	machineDeploymentLister machinelisters.MachineDeploymentLister
-	machinelisters          machinelisters.MachineLister
+	machineSetLister        machinelisters.MachineSetLister
```

Comment:
Why not nodeInformers too? It is used [below](https://github.com/gardener/autoscaler/pull/73/files#diff-fc87dde41e1a7abff0b0bc24dbabc258ac3d594a346b8f01e020b9e7f9775440R443).

Reply:
Good point. Something even thought about after going to sleep last night. 

I tried changing this today. However, was running into a vendoring mess. Let me see if i can fix this.

Reply:
As suggested, I have switched to nodeInformers [here](https://github.com/gardener/autoscaler/pull/73/commits/f8a13a52629f14399ead355ea4018a49a55a203b) as well. 

There is an optimization of reusing code here to use [these MCM libraries](https://github.com/gardener/machine-controller-manager/tree/master/pkg/util/clientbuilder) to build clients. However, while trying to make this change I am running into vendoring mess. The cause for the vendoring mess are two reasons, 
1. Mismatch in k8s versions used by MCM and Autoscaler. 
2. Go lang version on my system is `1.16` versus the required `1.15`. 

Both these points can be overcome using backs like rebuilding clients for MCM locally on 1.19 and switching the build on docker. However, in the interest of time and efforts I would like to leave out this optimization to post picking of [MCM k8s version update](https://github.com/gardener/machine-controller-manager/issues/402) which would ease solution this issue.

Reply:
Also have gotten rid of the coreClient as it is no longer used [here](https://github.com/gardener/autoscaler/pull/73/commits/f8a13a52629f14399ead355ea4018a49a55a203b#diff-fc87dde41e1a7abff0b0bc24dbabc258ac3d594a346b8f01e020b9e7f9775440L87)